src/test/management/list.spec.ts

```diff
@@ -119,4 +131,204 @@ describe("list", () => {
       );
     }
   });
+
+  describe("listFirebaseApps", () => {
+    it("should resolve with app list if it succeeds with only 1 api call", async () => {
+      const appCounts = 10;
+      const expectedAppList = generateAppList(appCounts, AppPlatform.PLATFORM_UNSPECIFIED);
+      const listAppsStub = listAppsApiRequestStub().resolves({
+        body: { apps: expectedAppList },
+      });
+
+      expect(await listFirebaseApps(PROJECT_ID)).to.deep.equal(expectedAppList);
+      expect(listAppsStub).to.be.calledOnce;
+    });
+
+    it("should resolve with iOS app list", async () => {
+      const appCounts = 10;
+      const expectedAppList = generateAppList(appCounts, AppPlatform.IOS);
+      const listAppsStub = listAppsApiRequestStub(AppPlatform.IOS).resolves({
+        body: { apps: expectedAppList },
+      });
+
+      expect(await listFirebaseApps(PROJECT_ID, AppPlatform.IOS)).to.deep.equal(expectedAppList);
+      expect(listAppsStub).to.be.calledOnce;
+    });
+
+    it("should resolve with Android app list", async () => {
+      const appCounts = 10;
+      const expectedAppList = generateAppList(appCounts, AppPlatform.ANDROID);
+      const listAppsStub = listAppsApiRequestStub(AppPlatform.ANDROID).resolves({
+        body: { apps: expectedAppList },
+      });
+
+      expect(await listFirebaseApps(PROJECT_ID, AppPlatform.ANDROID)).to.deep.equal(
+        expectedAppList
+      );
+      expect(listAppsStub).to.be.calledOnce;
+    });
+
+    it("should resolve with Web app list", async () => {
+      const appCounts = 10;
+      const expectedAppList = generateAppList(appCounts, AppPlatform.WEB);
+      const listAppsStub = listAppsApiRequestStub(AppPlatform.WEB).resolves({
+        body: { apps: expectedAppList },
+      });
+
+      expect(await listFirebaseApps(PROJECT_ID, AppPlatform.WEB)).to.deep.equal(expectedAppList);
+      expect(listAppsStub).to.be.calledOnce;
+    });
+
+    it("should concatenate pages to get app list if it succeeds", async () => {
+      const appCounts = 10;
+      const pageSize = 5;
+      const nextPageToken = "next-page-token";
+      const expectedAppList = generateAppList(appCounts, AppPlatform.PLATFORM_UNSPECIFIED);
+      const firstCallStub = listAppsApiRequestStub(undefined, pageSize).resolves({
```

Comment:
This usage of `listAppsApiRequestStub` makes me feel bad about the choice of the parameter `AppPlatform` being optional. This should take `PLATFORM_UNSPECIFIED`, rather than `undefined`. What does this mean for it to be `undefined`?

Reply:
Explained in an above comment that `undefined` means listing all apps.  I should have made the `expectedAppList` with arbitrary platforms here, requiring some concatenation of 3 `generateAppList` calls. Will update the tests for this